typings/index.d.ts

```diff
@@ -1846,7 +1846,7 @@ export class Shard extends EventEmitter {
   public ready: boolean;
   public worker: Worker | null;
   public eval(script: string): Promise<unknown>;
-  public eval<T>(fn: (client: Client) => T): Promise<T>;
+  public eval<T, C>(fn: (client: Client, context: C) => T, context: C): Promise<T>;
```

Comment:
`P` to stay consistent with the other methods.
And for `Serialized<T>` see (https://github.com/discordjs/discord.js/pull/5933)

```suggestion
  public eval<T, P>(fn: (client: Client, context: Serialized<P>) => T, context: P): Promise<T>;
```